Documentation/disaster-recovery.md

```diff
@@ -0,0 +1,112 @@
+# Disaster Recovery
+
+Self-hosted Kubernetes clusters are vulnerable to the following catastrophic
+failure scenarios:
+
+- Loss of all api-servers
+- Loss of all schedulers
+- Loss of all controller-managers
+- Loss of all self-hosted etcd nodes
+
+To minimize the likelihood of any of the these scenarios, production
+self-hosted clusters should always run in a [high-availability
+configuration](https://kubernetes.io/docs/admin/high-availability/).
+
+Nevertheless, in the event of a control plane loss the bootkube project
+provides limited disaster avoidance and recovery support through the
+`pod-checkpointer` program and the `bootkube recover` subcommand.
+
+## Pod Checkpointer
+
+The Pod Checkpointer is a program that ensures that existing local pod state
+can be recovered in the absence of an api-server.
+
+This is accomplished by managing "checkpoints" of local pod state as static pod
+manifests:
+
+- When the checkpointer sees that a "parent pod" (a pod which should be
+  checkpointed), is successfully running, the checkpointer will save a local
+  copy of the manifest.
+- If the parent pod is detected as no longer running, the checkpointer will
+  "activate" the checkpoint manifest. It will allow the checkpoint to continue
+  running until the parent-pod is restarted on the local node, or it is able to
+  contact an api-server to determine that the parent pod is no longer scheduled
+  to this node.
+
+A Pod Checkpointer DaemonSet is deployed by default when using `bootkube
+render` to create cluster manifests. Using the Pod Checkpointer is highly
+recommended for all self-hosted clusters to ensure node reboot resiliency.
+
+For more information, see the [Pod Checkpointer
+README](https://github.com/kubernetes-incubator/bootkube/blob/master/cmd/checkpoint/README.md).
+
+## Bootkube Recover
```

Comment:
We may want to have some kind of versioning convention. I'm assuming right now it's: you should always use the latest bootkube release when running recover. This may not be a confusion point, but I wonder if users will try and use the same bootkube release that they installed with (which is probably fine in most cases, unless there are new bug fixes they should have).